mesheryctl/internal/cli/root/design/view.go

```diff
@@ -36,6 +39,268 @@ var linkDocPatternView = map[string]string{
+					hasComponents = true
+					componentsCount = len(pfComponents)
+					components = pfComponents
+				}
+			}
+
+			// Check for relationships in pattern_file map
+			if pfRelsVal, exists := patternFile["relationships"]; exists && pfRelsVal != nil {
+				if pfRelationships, ok := pfRelsVal.([]interface{}); ok && len(pfRelationships) > 0 {
+					hasRelationships = true
+					relationshipsCount = len(pfRelationships)
+					relationships = pfRelationships
+				}
+			}
+		}
+
+		// Try pattern_file as string and parse JSON
+		if !hasComponents || !hasRelationships {
+			if pfStr, ok := patternFileVal.(string); ok && pfStr != "" {
+				var pfObj map[string]interface{}
+				if err := json.Unmarshal([]byte(pfStr), &pfObj); err == nil && pfObj != nil {
+					// Check for components in parsed pattern_file
+					if pfCompsVal, exists := pfObj["components"]; exists && pfCompsVal != nil {
+						if pfComponents, ok := pfCompsVal.([]interface{}); ok && len(pfComponents) > 0 {
+							hasComponents = true
+							componentsCount = len(pfComponents)
+							components = pfComponents
+						}
+					}
+
+					// Check for relationships in parsed pattern_file
+					if pfRelsVal, exists := pfObj["relationships"]; exists && pfRelsVal != nil {
+						if pfRelationships, ok := pfRelsVal.([]interface{}); ok && len(pfRelationships) > 0 {
+							hasRelationships = true
+							relationshipsCount = len(pfRelationships)
+							relationships = pfRelationships
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return hasComponents, componentsCount, components, hasRelationships, relationshipsCount, relationships
+}
+
+// displayComponentDetails formats and logs component details
+func displayComponentDetails(components []interface{}) {
+	if components == nil || len(components) == 0 {
+		return
+	}
+
+	utils.Log.Info("Component details:")
+	for i, compVal := range components {
+		comp, ok := compVal.(map[string]interface{})
+		if !ok || comp == nil {
+			utils.Log.Info(fmt.Sprintf("  %d. Unknown component (invalid format)", i+1))
+			continue
+		}
+
+		displayName := "Unknown"
+		kind := "Unknown"
+
+		if nameVal, exists := comp["displayName"]; exists && nameVal != nil {
+			if name, ok := nameVal.(string); ok {
+				displayName = name
+			} else {
+				displayName = fmt.Sprintf("%v", nameVal)
+			}
+		}
+
+		if compDataVal, exists := comp["component"]; exists && compDataVal != nil {
+			if compData, ok := compDataVal.(map[string]interface{}); ok && compData != nil {
+				if kindVal, exists := compData["kind"]; exists && kindVal != nil {
+					if k, ok := kindVal.(string); ok {
+						kind = k
+					} else {
+						kind = fmt.Sprintf("%v", kindVal)
+					}
+				}
+			}
+		}
+
+		utils.Log.Info(fmt.Sprintf("  %d. %s (Kind: %s)", i+1, displayName, kind))
+	}
+}
+
+// displayRelationshipDetails formats and logs relationship details
+func displayRelationshipDetails(relationships []interface{}) {
+	if relationships == nil || len(relationships) == 0 {
```

Comment:
```suggestion
	if len(relationships) == 0 {
```
len(nil) will return 0 and this will fix lint error